crypto/x509/x509_cmp.c

```diff
@@ -192,7 +192,7 @@ int ossl_x509_add_cert_new(STACK_OF(X509) **p_sk, X509 *cert, int flags)
 
 int X509_add_cert(STACK_OF(X509) *sk, X509 *cert, int flags)
 {
-    if (sk == NULL) {
+    if (sk == NULL || cert == NULL) {
         ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
         return 0;
     }
```

Comment:
Thanks for the fix!

I have two comments:

1.
Refering to the man page:
https://docs.openssl.org/master/man3/X509_add_cert/

> X509_add_cert() adds a certificate cert to the given list sk.
> 
> X509_add_certs() adds a list of certificate certs to the given list sk. The certs argument may be NULL, which implies no effect. It does not modify the list certs but in case the X509_ADD_FLAG_UP_REF flag (described below) is set the reference counters of those of its members added to sk are increased.

I would expect the `X509_add_cert()` to handle NULL cert similarly to NULL certs. I.e. return success.

Suggested change:
```
    if (sk == NULL) {
        ERR_raise(ERR_LIB_X509, ERR_R_PASSED_NULL_PARAMETER);
        return 0;
    }

    if (cert == NULL)
        return 1;
```

2.
Could you please update the man page (doc/man3/X509_add_cert.pod) with the new behaviour?

I don't know if there are any compatibility concerns with changing the behaviour for NULL cert?